### PR TITLE
Feature/3x3 simple

### PR DESCRIPTION
**Pull request by Raycoms**
Don't merge this yet. WIP

# Changes proposed in this pull request:
- Simple 3x3 crafting implementable for workers
- Fixes scanning and replacing double slabs
- Fixes scan tool in smp.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import com.minecolonies.api.colony.requestsystem.requestable.Stack;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java]: ### Errors:
- Unused import import com.minecolonies.api.colony.requestsystem.requestable.Stack;

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
...
 import com.minecolonies.api.colony.requestsystem.token.IToken;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java]: ### Errors:
- Unused import import com.minecolonies.api.colony.requestsystem.token.IToken;

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java [hunk 3]
```diff
@@ -9,6 +9,7 @@
...
 import com.minecolonies.api.crafting.IRecipeStorage;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java]: ### Errors:
- Unused import import com.minecolonies.api.crafting.IRecipeStorage;

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java [hunk 4]
```diff
@@ -9,6 +9,7 @@
...
+import com.minecolonies.api.crafting.ItemStorage;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java]: ### Errors:
- Unused import import com.minecolonies.api.crafting.ItemStorage;

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java [hunk 5]
```diff
@@ -9,6 +9,7 @@
...
 import com.minecolonies.api.util.ItemStackUtils;
```
Comment (LDTTeam-Buildsystem) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java]: ### Errors:
- Unused import import com.minecolonies.api.util.ItemStackUtils;

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |  662  |  0  |  0  |  0   |
| Errors |   8   |  0  |  0  |  0   |

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 33865 |  5  |  6  |  -1  |
| Errors |  591  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=11191&tab=Inspection&buildTypeId=bt88)

**Comment by winex** [comment]
this is cool, but isn't it easier to replace current 2x2 grid with 3x3 and just disable slots in GUI (and, of course, check for the bad input on the server side)?
also, put some note like _"this type of worker unable to craft 3x3"_ or _"upgrade building/worker is too dumb for smth la-la-la"_ :)
why so complex, so many lines of code?

**Comment by marchermans** [comment]
Lets just say this is a small part of a much bigger machine.

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New  | Old  | Diff |
| ------ |:-----:|:----:|:----:|:----:|
| Total  | 34242 | 1555 | 1468 | +87  |
| Errors |  592  |  0   |  0   |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=12740&tab=Inspection&buildTypeId=bt88)

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New  | Old  | Diff |
| ------ |:-----:|:----:|:----:|:----:|
| Total  | 34242 | 1555 | 1468 | +87  |
| Errors |  592  |  0   |  0   |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=12740&tab=Inspection&buildTypeId=bt88)

**Comment by winex** [comment]
just curious, what gone wrong with "Fix baker wheat request"?

**Comment by Raycoms** [comment]
I didn't notice that the other pr fixed this issue already.

2018-07-19 15:00 GMT-03:00 Roman Vasiyarov <notifications@github.com>:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New  | Old  | Diff  |
| ------ |:-----:|:----:|:----:|:-----:|
| Total  | 38307 | 9915 | 5850 | +4065 |
| Errors |  592  |  0   |  0   |  +0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=13625&tab=Inspection&buildTypeId=bt88)

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 38436 | 577 | 448 | +129 |
| Errors |  592  |  0  |  0  |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=13997&tab=Inspection&buildTypeId=bt88)

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New  | Old  | Diff  |
| ------ |:-----:|:----:|:----:|:-----:|
| Total  | 40685 | 4890 | 2641 | +2249 |
| Errors |  675  |  93  |  10  |  +83  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=17302&tab=Inspection&buildTypeId=bt88)

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40684 |  2  |  3  |  -1  |
| Errors |  675  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=17368&tab=Inspection&buildTypeId=bt88)

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40684 |  2  |  3  |  -1  |
| Errors |  675  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=17368&tab=Inspection&buildTypeId=bt88)

File: src/main/java/com/minecolonies/coremod/compatibility/jei/transer/PrivateCraftingTeachingTransferHandler.java [hunk 6]
```diff
@@ -79,20 +94,37 @@ public IRecipeTransferError transferRecipe(
...
             }
-        }, 2, 2);
+        }, size, size);
 
-        craftMatrix.setInventorySlotContents(0, guiIngredients.get(0));
-        craftMatrix.setInventorySlotContents(1, guiIngredients.get(1));
-        craftMatrix.setInventorySlotContents(2, guiIngredients.get(3));
-        craftMatrix.setInventorySlotContents(3, guiIngredients.get(4));
 
+        if(craftingGUIBuilding.isComplete())
```
Comment (Nightenom) [inline src/main/java/com/minecolonies/coremod/compatibility/jei/transer/PrivateCraftingTeachingTransferHandler.java:108]: What about excluding the first four as they will be executed every time?
_more times in this pr_

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/compatibility/jei/transer/PrivateCraftingTeachingTransferHandler.java:108]: On purpose, the index is slightly different

**Review by LDTTeam-Buildsystem — APPROVED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Review by LDTTeam-Buildsystem — APPROVED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  |   0   |  0  |  0  |  0   |
| Errors |   0   |  0  |  0  |  0   |
##### More Information:

**Pull request merged**